app/windows/codesign.js

```diff
@@ -0,0 +1,166 @@
+const { exec } = require('child_process');
+const path = require('path');
+const os = require('os');
+const fs = require('fs');
+
+const SIGN_JSON_TEMPLATE = {
+  Version: '1.0.0',
+  DriEmail: ['joaquim.rocha@microsoft.com'],
+  GroupId: null,
+  CorrelationVector: null,
+  SignBatches: [],
+};
+
+const POLICY_JSON = {
+  Version: '1.0.0',
+  Intent: '',
+  ContentType: '',
+  ContentOrigin: '',
+  ProductState: '',
+  Audience: '',
+};
+
+const AUTH_JSON = {
+  Version: '1.0.0',
+  AuthenticationType: 'AAD_CERT',
+  ClientId: `${process.env.HEADLAMP_WINDOWS_CLIENT_ID}`,
+  AuthCert: {
+    SubjectName: `CN=${process.env.HEADLAMP_WINDOWS_CLIENT_ID}.microsoft.com`,
+    StoreLocation: 'LocalMachine',
+    StoreName: 'My',
+    SendX5c: 'true',
+  },
+  RequestSigningCert: {
+    SubjectName: `CN=${process.env.HEADLAMP_WINDOWS_CLIENT_ID}`,
+    StoreLocation: 'LocalMachine',
+    StoreName: 'My',
+  },
+};
+
+function getFileList(rootDir) {
+  let files = {};
+  let dirs = ['.'];
+  while (dirs.length > 0) {
+    const dirName = dirs.shift();
+    const curDir = path.join(rootDir, dirName);
+
+    fs.readdirSync(curDir).forEach(file => {
+      if (file === 'node_modules') {
```

Comment:
Any other folders to skip? I guess .git isn't in here.

Reply:
Right. This may require fine tuning later. For now the signing goes file by file, but it's useful to be able to generate the signing for an entire dir like this.